rust/arrow/src/compute/kernels/filter.rs

```diff
@@ -424,7 +424,7 @@ impl FilterContext {
     where
         T: ArrowNumericType,
     {
-        let array_type = T::get_data_type();
+        let array_type = T::DATA_TYPE;
```

Comment:
this could actually be removed by making `filter_array_impl` a generic over `T`. I left for a future exercise.